interfaces/builtin/cifs_mount_control.go

```diff
@@ -0,0 +1,71 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const cifsMountControlSummary = `allows to mount and unmount CIFS shares`
+
+const cifsMountControlBaseDeclarationSlots = `
+  cifs-mount-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const cifsMountControlConnectedPlugSecComp = `
+# Description: Allow mount and umount syscall access.
+
+mount
+umount
+umount2
+`
+
+const cifsMountControlConnectedPlugAppArmor = `
+# Description: Allow to mount and unmount CIFS filesystems.
+
+# Required for mounts and unmounts
+capability sys_admin,
+
+# Allow mounts to our snap-specific writable directories
+mount fstype=cifs ** -> /home/*/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**/},
+mount fstype=cifs ** -> /var/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**/},
+mount fstype=cifs ** -> /home/*/snap/@{SNAP_NAME}/common/{,**/},
+mount fstype=cifs ** -> /var/snap/@{SNAP_NAME}/common/{,**/},
+
+umount fstype=cifs /home/*/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**/},
+umount fstype=cifs /var/snap/@{SNAP_NAME}/@{SNAP_REVISION}/{,**/},
+umount fstype=cifs /home/*/snap/@{SNAP_NAME}/common/{,**/},
+umount fstype=cifs /var/snap/@{SNAP_NAME}/common/{,**/},
+
+/run/mount/utab rw,
```

Comment:
Not sure if we want this since the filesystem is not really visibly mounted from the outside.

Reply:
Without the /run/mount/utab permission the mount (or umount?) command did fail. It showed up in the apparmor log that's why I added it. But I am happy for other suggestions. I am not too deep into apparmor / snap container confinements. So any other suggestion is welcome.

Reply:
Well the problem is that /run/mount/utab is _shared_ with the host. So on the outside it will _seem_ from that file alone that some CIFS is mounted but when you go and look (again on the outside of snap application's point of view) they are not.

Does it fail outright if you drop this rule?

Reply:
If I remember correctly, yes. But I will retry this later and record the logs and post them here, just to be sure.

Just in case this is relevant somehow: When I executed "mount" outside of the snap the shares were not visible in the output, when I entered the snaps name space with "snap run" I was able to see them in the list.

Reply:
That's true (because the mount namespace shields us) but the userspace libmount library does use that file for extra bookkeeping. In some cases it can clash with what is really mounted outside so I think this needs some more investigation.